### DISCUSSION TITLE
ISG not working on Google Cloud Platform

### DISCUSSION THREAD

**Discussion by timothyarmes** [vercel/next.js#31444] [Help]
### What version of Next.js are you using?

12.0.0

### What version of Node.js are you using?

16.12.0

### What browser are you using?

Chrome

### What operating system are you using?

Linux (running on GCP)

### How are you deploying your application?

next start

### Describe the Bug

I have a Next.js server running on Google Cloud Platform, and the ISG feature isn't working, the pages are never updated.

The only hint that I have in the logs is this error:

`Container Sandbox: Unsupported syscall sched_getattr(0x1f,0x3e9c0c1f1620,0x38,0x0,0x3e9c0c1f17f0,0x3e9c12c725e0). It is very likely that you can safely ignore this message and that this is not the cause of any error you might be troubleshooting. Please, refer to https://gvisor.dev/docs/user_guide/compatibility/linux/amd64/sched_getattr for more information.`

Despite the message I'm suspicious since that sys call is scheduling related,  but at the same time it would be surprising that NEXT is using something obscure to handle this.

Has anyone else experienced this issue?

My docker file is really simple:

```
FROM node:16

WORKDIR /app

COPY package*.json .
RUN yarn install

COPY . .

RUN yarn build

ENTRYPOINT [ "yarn", "start" ]
```

where `yarn start` starts up Next like this:

```
const compression = require('compression');
const express = require('express');
const next = require('next');

const port = parseInt(process.env.PORT, 10) || 80;
const dev = process.env.NODE_ENV !== 'production';
const app = next({ dev, dir: __dirname });
const handle = app.getRequestHandler();
const { Sentry } = require('./services/sentry');

app.prepare().then(() => {
  const server = express();

  server.use(Sentry.Handlers.requestHandler());
  server.use(compression());

  server.get('/health', (req, res) => {
    return res.json({ status: 'pass' });
  });

  server.get('*', (req, res) => {
    return handle(req, res);
  });

  server.use(Sentry.Handlers.errorHandler());

  server.listen(port, err => {
    if (err) throw err;
    console.log(`> Ready on http://localhost:${port}`);
  });
});
```

If I build and run locally then the ISG works.

### Expected Behavior

I'd expect the ISG pages to be rerendered

### To Reproduce

Put an ISG based test on GCP???

**Comment by timneutkens** [comment]
Converted to discussions given that the application code was not provided. It's likely that your getStaticProps function is throwing an error, which means Next.js will reuse the already generated page to ensure it does not break your app.

**Reply by timothyarmes** [reply]
@timneutkens In that case I would expect the logs to display a message, and they're not.  Also, if I build and start locally it works.
